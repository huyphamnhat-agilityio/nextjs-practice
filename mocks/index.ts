export * from "./category";
export * from "./product";
export * from "./fetch";
export * from "./avatar";
export * from "./image";
