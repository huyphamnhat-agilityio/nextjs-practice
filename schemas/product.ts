import { z } from "zod";

// Constants
import {
  ACCEPTED_IMAGE_TYPES,
  FORM_MESSAGES,
  MAX_IMAGE_FILE_SIZE,
} from "@/constants";

export const ProductFormSchema = z
  .object({
    id: z.string(),
    category: z.string({
      required_error: FORM_MESSAGES.PRODUCT.CATEGORY.REQUIRED,
    }),
    title: z
      .string({ required_error: FORM_MESSAGES.PRODUCT.TITLE.REQUIRED })
      .min(2, FORM_MESSAGES.PRODUCT.TITLE.MIN)
      .max(100, FORM_MESSAGES.PRODUCT.TITLE.MAX),
    description: z
      .string()
      .max(200, FORM_MESSAGES.PRODUCT.DESCRIPTION.MAX)
      .optional(),
    sales: z
      .number({ required_error: FORM_MESSAGES.PRODUCT.SALES.REQUIRED })
      .min(0, FORM_MESSAGES.PRODUCT.SALES.MIN),
    originalPrice: z
      .number({ required_error: FORM_MESSAGES.PRODUCT.ORIGINAL_PRICE.REQUIRED })
      .min(0, FORM_MESSAGES.PRODUCT.ORIGINAL_PRICE.MIN),
    salePrice: z
      .number({ required_error: FORM_MESSAGES.PRODUCT.SALE_PRICE.REQUIRED })
      .min(0, FORM_MESSAGES.PRODUCT.SALE_PRICE.MIN),
    rate: z
      .number({ required_error: FORM_MESSAGES.PRODUCT.RATE.REQUIRED })
      .min(0, FORM_MESSAGES.PRODUCT.RATE.MIN)
      .max(5, FORM_MESSAGES.PRODUCT.RATE.MAX)
      .refine(
        (val) => {
          return /^\d+(\.\d)?$/.test(val.toString());
        },
        {
          message: FORM_MESSAGES.PRODUCT.RATE.DECIMAL,
        },
      ),
    coverImage: z
      .any()
      .optional()
      .refine(
        (file) => !file || file?.size <= MAX_IMAGE_FILE_SIZE,
        FORM_MESSAGES.PRODUCT.COVER_IMAGE.MAX_SIZE,
      )
      .refine(
        (file) => !file || ACCEPTED_IMAGE_TYPES.includes(file?.type),
        FORM_MESSAGES.PRODUCT.COVER_IMAGE.ACCEPTED_FORMATS,
      ),
  })
  .refine((schema) => schema.originalPrice >= schema.salePrice, {
    message: FORM_MESSAGES.PRODUCT.SALE_PRICE.MAX,
    path: ["salePrice"],
  })
  .refine((schema) => schema.originalPrice >= schema.salePrice, {
    message: FORM_MESSAGES.PRODUCT.ORIGINAL_PRICE.CONSTRAINT,
    path: ["originalPrice"],
  });
