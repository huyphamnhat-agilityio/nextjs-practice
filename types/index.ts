export * from "./navbar";
export * from "./product";
export * from "./category";
