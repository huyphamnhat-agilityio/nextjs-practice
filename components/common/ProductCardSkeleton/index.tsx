import {
  Card,
  CardBody,
  CardHeader,
  Image as NextUIImage,
  Skeleton,
} from "@nextui-org/react";

// Components
import { Button } from "../Button";
import {
  CartIcon,
  DownloadIcon,
  EyeIcon,
  FavoriteIcon,
  NextArrowIcon,
  SolidStarIcon,
} from "../Icons";

// types
import { Product } from "@/types";

const ProductCardSkeleton = ({
  category,
  description,
  rate,
  originalPrice,
  salePrice,
  title,
  sales,
}: Product) => (
  <Card className="max-w-82 flex flex-col rounded-none">
    <CardHeader className="relative p-0">
      <Skeleton>
        <NextUIImage width={328} height={300} radius="none" />
      </Skeleton>
      <div className="px-2.5 bg-danger absolute rounded-0.75 top-5 left-5 z-10">
        <p className="text-white text-sm/6 font-bold">Sale</p>
      </div>
      <div className="flex gap-2.5 absolute bottom-6 left-24 z-10">
        <Button className="bg-white" variant="action" size="tiny" isIconOnly>
          <FavoriteIcon />
        </Button>
        <Button className="bg-white" variant="action" size="tiny" isIconOnly>
          <CartIcon />
        </Button>
        <Button className="bg-white" variant="action" size="tiny" isIconOnly>
          <EyeIcon />
        </Button>
      </div>
    </CardHeader>
    <CardBody className="light px-6.25 pt-6.25 pb-8.75 bg-white flex flex-col gap-2.5">
      <div className="flex justify-between">
        <Skeleton>
          <span className="text-primary text-sm/6 font-bold">{category}</span>
        </Skeleton>
        <Skeleton>
          <div className="p-1.25 flex items-center gap-1.25 bg-dark-blue rounded-[20px]">
            <SolidStarIcon />
            <span className="text-white text-xs">{rate}</span>
          </div>
        </Skeleton>
      </div>
      <Skeleton>
        <h5 className="text-foreground text-base font-bold">{title}</h5>
      </Skeleton>
      <Skeleton>
        <p className="text-foreground-100 text-sm">{description}</p>
      </Skeleton>
      <Skeleton>
        <div className="flex items-center gap-2.5 text-foreground-100 text-sm/6 font-bold">
          <DownloadIcon />
          {sales} Sales
        </div>
      </Skeleton>
      <Skeleton>
        <div className="py-1.25 px-0.75 flex gap-1.25">
          <span className="text-foreground-50 text-base font-bold">
            {originalPrice}
          </span>
          <span className="text-secondary text-base font-bold">
            {salePrice}
          </span>
        </div>
      </Skeleton>
      <div>
        <Button
          color="primary"
          variant="bordered"
          size="xs"
          endContent={<NextArrowIcon />}
          className="font-bold"
        >
          Learn More
        </Button>
      </div>
    </CardBody>
  </Card>
);

export default ProductCardSkeleton;
