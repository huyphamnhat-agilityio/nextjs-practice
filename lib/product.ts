"use server";
import { fetchApi } from "./fetch";

// Types
import { Pagination, Product } from "@/types";

// Constants
import { RESOURCES } from "@/constants";
import { revalidatePath } from "next/cache";

export type GetProductProps = {
  page: number;
  limit: number;
  query?: string;
};
export const getProducts = async ({
  page,
  limit,
  query = "",
}: GetProductProps) => {
  const options = {
    publicDataUrl: `${process.env.MOCK_API}/${RESOURCES.PRODUCT}?title=${query}`,
    page,
    limit,
  };

  try {
    const products = await fetchApi<Pagination<Product>>(
      process.env.PAGINATION_API,
      {
        method: "POST",
        body: JSON.stringify(options),
        next: {
          revalidate: 300,
        },
      },
    );

    return products;
  } catch (error) {
    if (error.status === 404 || error.status === 400) return undefined;
    throw new Error(error.message);
  }
};

export const markProduct = async (data: Product) => {
  try {
    const response = await fetchApi<Product>(
      `${process.env.MOCK_API}/${RESOURCES.PRODUCT}/${data.id}`,
      {
        method: "PUT",
        body: JSON.stringify({ ...data, isFavorited: !data.isFavorited }),
      },
    );
    revalidatePath("/", "layout");

    return response;
  } catch (error) {
    throw error;
  }
};

export const createProduct = async (data: FormData) => {
  console.log(data.get("description"));
};
