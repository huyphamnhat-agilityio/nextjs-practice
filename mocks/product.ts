import { MOCK_CATEGORIES } from "./category";

import { Pagination, Product } from "@/types";

export const MOCK_PRODUCT_DESCRIPTION =
  "We focus on ergonomics and meeting you where you work. It's only a keystroke away.";

export const MOCK_PRODUCTS: Array<Product> = [
  {
    id: "1",
    category: MOCK_CATEGORIES[0].name,
    rate: 4.9,
    title: "Video in Live Action",
    description: MOCK_PRODUCT_DESCRIPTION,
    sales: 0,
    originalPrice: 16.48,
    salePrice: 6.48,
    coverImageUrl: "/product-cover-1.webp",
    isFavorited: 0,
    createdAt: 0,
  },
  {
    id: "2",
    category: MOCK_CATEGORIES[1].name,
    rate: 4.9,
    title: "Every Client Matters",
    description: MOCK_PRODUCT_DESCRIPTION,
    sales: 15,
    originalPrice: 16.48,
    salePrice: 6.48,
    coverImageUrl: "/product-cover-2.webp",
    isFavorited: 1,
    createdAt: 10,
  },
  {
    id: "3",
    category: MOCK_CATEGORIES[2].name,
    rate: 4.9,
    title: "Get Quality Education",
    description: MOCK_PRODUCT_DESCRIPTION,
    sales: 15,
    originalPrice: 16.48,
    salePrice: 6.48,
    coverImageUrl: "/product-cover-3.webp",
    isFavorited: 0,
    createdAt: 0,
  },
  {
    id: "4",
    category: MOCK_CATEGORIES[0].name,
    rate: 4.9,
    title: "Video in Live Action",
    description: MOCK_PRODUCT_DESCRIPTION,
    sales: 15,
    originalPrice: 16.48,
    salePrice: 6.48,
    coverImageUrl: "/product-cover-1.webp",
    isFavorited: 0,
    createdAt: 0,
  },
  {
    id: "5",
    category: MOCK_CATEGORIES[1].name,
    rate: 4.9,
    title: "Every Client Matters",
    description: MOCK_PRODUCT_DESCRIPTION,
    sales: 15,
    originalPrice: 16.48,
    salePrice: 6.48,
    coverImageUrl: "/product-cover-2.webp",
    isFavorited: 1,
    createdAt: 10,
  },
  {
    id: "6",
    category: MOCK_CATEGORIES[2].name,
    rate: 4.9,
    title: "Get Quality Education",
    description: MOCK_PRODUCT_DESCRIPTION,
    sales: 15,
    originalPrice: 16.48,
    salePrice: 6.48,
    coverImageUrl: "/product-cover-3.webp",
    isFavorited: 0,
    createdAt: 0,
  },
];

export const PLACEHOLDER_PRODUCT_FORM_DATA = {
  id: "",
  category: "",
  title: "",
  description: "",
  sales: undefined,
  originalPrice: undefined,
  salePrice: undefined,
  rate: undefined,
  coverImageUrl: "",
  coverImage: undefined,
  isFavorited: 0,
  createdAt: 0,
};

export const PLACEHOLDER_PRODUCT_DESCRIPTION = "This course has no description";
