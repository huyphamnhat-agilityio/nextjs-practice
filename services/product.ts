import { fetchApi } from "./fetch";

// Types
import { Pagination, Product } from "@/types";

// Constants
import { RESOURCES } from "@/constants";

export const getProducts = async (page: number, limit: number) => {
  const options = {
    publicDataUrl: `${process.env.MOCK_API}/${RESOURCES.PRODUCT}`,
    page,
    limit,
  };

  try {
    const products = await fetchApi<Pagination<Product>>(
      process.env.PAGINATION_API,
      {
        method: "POST",
        body: JSON.stringify(options),
      },
    );

    return products;
  } catch (error) {
    console.error("Fetch Error:", error);
    throw new Error("Failed to fetch product data.");
  }
  // try {
  //   const products = await fetchApi<Array<Product>>(
  //     `${process.env.MOCK_API}/${RESOURCES.PRODUCT}`
  //   );

  //   console.log(products);
  // } catch (error) {
  //   console.error("Fetch Error:", error);
  //   throw new Error("Failed to fetch product data.");
  // }
};
