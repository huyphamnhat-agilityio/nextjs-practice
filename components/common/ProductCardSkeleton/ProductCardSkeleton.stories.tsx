import type { Meta, StoryObj } from "@storybook/react";

// Components
import ProductCardSkeleton from ".";

// Mocks
import { MOCK_PRODUCTS } from "@/mocks/product";

const meta: Meta<typeof ProductCardSkeleton> = {
  title: "Components/common/ProductCardSkeleton",
  component: ProductCardSkeleton,
  argTypes: {
    id: {
      description: "Product ID",
    },
    categoryId: {
      description: "Product category ID",
    },
    category: {
      description: "Product category",
    },
    description: {
      description: "Product description",
    },
    coverImageUrl: {
      description: "Product cover image URL",
    },
    originalPrice: {
      description: "Product original price",
    },
    salePrice: {
      description: "Product sale price",
    },
    rate: {
      description: "Product rating",
    },
    title: {
      description: "Product title",
    },
  },
  parameters: {
    controls: {
      expanded: true,
    },
  },
};

export default meta;

type Story = StoryObj<typeof ProductCardSkeleton>;

export const Primary: Story = {
  args: { ...MOCK_PRODUCTS[0] },
};
