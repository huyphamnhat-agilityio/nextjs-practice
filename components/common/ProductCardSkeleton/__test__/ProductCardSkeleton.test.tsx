import { render } from "@testing-library/react";

// Components
import ProductCardSkeleton from "..";

// Types
import { Product } from "@/types";

// Mocks
import { MOCK_PRODUCTS } from "@/mocks/product";

describe("ProductCarkSkeleton test cases", () => {
  const setup = (props: Product) => render(<ProductCardSkeleton {...props} />);

  it("should render correctly", () => {
    const { asFragment } = setup(MOCK_PRODUCTS[0]);

    expect(asFragment()).toMatchSnapshot();
  });
});
