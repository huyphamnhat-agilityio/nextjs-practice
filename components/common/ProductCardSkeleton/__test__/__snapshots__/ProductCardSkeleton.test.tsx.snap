// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCarkSkeleton test cases should render correctly 1`] = `
<DocumentFragment>
  <div
    class="relative overflow-hidden h-auto text-foreground box-border bg-content1 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium transition-transform-background motion-reduce:transition-none max-w-82 flex flex-col rounded-none"
    tabindex="-1"
  >
    <div
      class="flex z-10 w-full justify-start items-center shrink-0 overflow-inherit color-inherit subpixel-antialiased rounded-t-large relative p-0"
    >
      <div
        class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 before:transition-opacity before:!duration-300"
      >
        <div
          class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
        >
          <div
            class="relative shadow-black/5 shadow-none rounded-none"
            style="max-width: 328px;"
          >
             
            <img
              class="relative z-10 opacity-0 shadow-black/5 data-[loaded=true]:opacity-100 shadow-none transition-transform-opacity motion-reduce:transition-none !duration-300 rounded-none"
              height="300"
              style="height: 300px;"
              width="328"
            />
          </div>
        </div>
      </div>
      <div
        class="px-2.5 bg-danger absolute rounded-0.75 top-5 left-5 z-10"
      >
        <p
          class="text-white text-sm/6 font-bold"
        >
          Sale
        </p>
      </div>
    </div>
    <div
      class="relative w-full p-3 flex-auto place-content-inherit align-items-inherit h-auto break-words text-left overflow-y-auto subpixel-antialiased light px-6.25 pt-6.25 pb-8.75 bg-white flex flex-col gap-2.5"
    >
      <div
        class="flex justify-between"
      >
        <div
          class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 before:transition-opacity before:!duration-300"
        >
          <div
            class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
          >
            <span
              class="text-primary text-sm/6 font-bold"
            >
              Training Courses
            </span>
          </div>
        </div>
        <div
          class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 before:transition-opacity before:!duration-300"
        >
          <div
            class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
          >
            <div
              class="p-1.25 flex items-center gap-1.25 bg-dark-blue rounded-[20px]"
            >
              <svg
                fill="none"
                height="23"
                viewBox="0 0 23 23"
                width="23"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20.6369 8.41964C20.5793 8.24983 20.4731 8.10061 20.3317 7.99043C20.1902 7.88024 20.0195 7.81389 19.8408 7.79959L14.6116 7.38408L12.3488 2.37506C12.2767 2.21373 12.1595 2.07671 12.0113 1.98052C11.8631 1.88434 11.6902 1.8331 11.5135 1.83301C11.3368 1.83291 11.1638 1.88395 11.0155 1.97998C10.8672 2.076 10.7498 2.2129 10.6776 2.37415L8.41477 7.38408L3.18562 7.79959C3.00993 7.8135 2.84196 7.87774 2.70183 7.98462C2.56169 8.0915 2.45531 8.23648 2.39541 8.40224C2.33552 8.56799 2.32464 8.74749 2.36409 8.91926C2.40354 9.09103 2.49164 9.2478 2.61785 9.37081L6.48216 13.1379L5.11548 19.0559C5.07398 19.235 5.08728 19.4225 5.15365 19.594C5.22003 19.7655 5.33641 19.913 5.48769 20.0175C5.63898 20.1221 5.81819 20.1787 6.00206 20.1801C6.18594 20.1815 6.36598 20.1276 6.51885 20.0254L11.5132 16.6958L16.5075 20.0254C16.6638 20.1291 16.848 20.1825 17.0355 20.1784C17.223 20.1743 17.4047 20.1129 17.5563 20.0024C17.7078 19.8919 17.8219 19.7377 17.8832 19.5604C17.9444 19.3832 17.9499 19.1914 17.899 19.0109L16.2213 13.1406L20.3819 9.39649C20.6543 9.15067 20.7543 8.76727 20.6369 8.41964Z"
                  fill="#F3CD03"
                />
              </svg>
              <span
                class="text-white text-xs"
              >
                4.9
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 before:transition-opacity before:!duration-300"
      >
        <div
          class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
        >
          <h5
            class="text-foreground text-base font-bold"
          >
            Video in Live Action
          </h5>
        </div>
      </div>
      <div
        class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 before:transition-opacity before:!duration-300"
      >
        <div
          class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
        >
          <p
            class="text-foreground-100 text-sm"
          >
            We focus on ergonomics and meeting you where you work. It's only a keystroke away.
          </p>
        </div>
      </div>
      <div
        class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 before:transition-opacity before:!duration-300"
      >
        <div
          class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
        >
          <div
            class="flex items-center gap-2.5 text-foreground-100 text-sm/6 font-bold"
          >
            <svg
              fill="none"
              height="16"
              viewBox="0 0 17 16"
              width="17"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0_2685_1042)"
              >
                <path
                  d="M1 9.90039C1.13261 9.90039 1.25979 9.95307 1.35355 10.0468C1.44732 10.1406 1.5 10.2678 1.5 10.4004V12.9004C1.5 13.1656 1.60536 13.42 1.79289 13.6075C1.98043 13.795 2.23478 13.9004 2.5 13.9004H14.5C14.7652 13.9004 15.0196 13.795 15.2071 13.6075C15.3946 13.42 15.5 13.1656 15.5 12.9004V10.4004C15.5 10.2678 15.5527 10.1406 15.6464 10.0468C15.7402 9.95307 15.8674 9.90039 16 9.90039C16.1326 9.90039 16.2598 9.95307 16.3536 10.0468C16.4473 10.1406 16.5 10.2678 16.5 10.4004V12.9004C16.5 13.4308 16.2893 13.9395 15.9142 14.3146C15.5391 14.6897 15.0304 14.9004 14.5 14.9004H2.5C1.96957 14.9004 1.46086 14.6897 1.08579 14.3146C0.710714 13.9395 0.5 13.4308 0.5 12.9004V10.4004C0.5 10.2678 0.552678 10.1406 0.646447 10.0468C0.740215 9.95307 0.867392 9.90039 1 9.90039Z"
                  fill="#737373"
                />
                <path
                  d="M8.14566 11.854C8.1921 11.9006 8.24728 11.9375 8.30802 11.9627C8.36877 11.9879 8.43389 12.0009 8.49966 12.0009C8.56542 12.0009 8.63054 11.9879 8.69129 11.9627C8.75203 11.9375 8.80721 11.9006 8.85366 11.854L11.8537 8.854C11.9475 8.76011 12.0003 8.63278 12.0003 8.5C12.0003 8.36722 11.9475 8.23989 11.8537 8.146C11.7598 8.05211 11.6324 7.99937 11.4997 7.99937C11.3669 7.99937 11.2395 8.05211 11.1457 8.146L8.99966 10.293V1.5C8.99966 1.36739 8.94698 1.24021 8.85321 1.14645C8.75944 1.05268 8.63226 1 8.49966 1C8.36705 1 8.23987 1.05268 8.1461 1.14645C8.05233 1.24021 7.99966 1.36739 7.99966 1.5V10.293L5.85366 8.146C5.75977 8.05211 5.63243 7.99937 5.49966 7.99937C5.36688 7.99937 5.23954 8.05211 5.14566 8.146C5.05177 8.23989 4.99902 8.36722 4.99902 8.5C4.99902 8.63278 5.05177 8.76011 5.14566 8.854L8.14566 11.854Z"
                  fill="#737373"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_2685_1042"
                >
                  <rect
                    fill="white"
                    height="16"
                    transform="translate(0.5)"
                    width="16"
                  />
                </clippath>
              </defs>
            </svg>
            15 Sales
          </div>
        </div>
      </div>
      <div
        class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 before:transition-opacity before:!duration-300"
      >
        <div
          class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
        >
          <div
            class="py-1.25 px-0.75 flex gap-1.25"
          >
            <span
              class="text-foreground-50 text-base font-bold"
            >
              16.48
            </span>
            <span
              class="text-secondary text-base font-bold"
            >
              6.48
            </span>
          </div>
        </div>
      </div>
      <div>
        <button
          class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-medium text-small gap-2 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-transparent border-primary text-primary data-[hover=true]:opacity-hover px-5 py-2.5 min-w-0 w-auto h-auto rounded-[37px] hover:bg-primary hover:bg-opacity-25 font-bold"
          type="button"
        >
          Course Detail
          <svg
            fill="none"
            height="17"
            viewBox="0 0 10 17"
            width="10"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clip-path="url(#clip0_2685_983)"
            >
              <path
                clip-rule="evenodd"
                d="M0.680771 1.00694C0.737928 0.949641 0.805828 0.904178 0.880583 0.873159C0.955337 0.842139 1.03548 0.826172 1.11641 0.826172C1.19735 0.826172 1.27749 0.842139 1.35224 0.873159C1.427 0.904178 1.4949 0.949641 1.55205 1.00694L8.9358 8.39069C8.9931 8.44785 9.03857 8.51575 9.06959 8.5905C9.10061 8.66526 9.11657 8.7454 9.11657 8.82633C9.11657 8.90727 9.10061 8.98741 9.06959 9.06216C9.03857 9.13692 8.9931 9.20482 8.9358 9.26197L1.55205 16.6457C1.43651 16.7613 1.27981 16.8262 1.11641 16.8262C0.953015 16.8262 0.79631 16.7613 0.680771 16.6457C0.565232 16.5302 0.500322 16.3735 0.500322 16.2101C0.500322 16.0467 0.565232 15.89 0.680771 15.7744L7.63011 8.82633L0.680771 1.87823C0.623469 1.82107 0.578006 1.75317 0.546986 1.67841C0.515967 1.60366 0.5 1.52352 0.5 1.44258C0.5 1.36165 0.515967 1.28151 0.546986 1.20676C0.578006 1.132 0.623469 1.0641 0.680771 1.00694Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </g>
            <defs>
              <clippath
                id="clip0_2685_983"
              >
                <rect
                  fill="currentColor"
                  height="16"
                  transform="translate(0.5 0.826172)"
                  width="9"
                />
              </clippath>
            </defs>
          </svg>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
